go/dhcp/config.go

```diff
@@ -340,6 +355,15 @@ func (d *Interfaces) readConfig() {
 						options[dhcp.OptionDomainNameServer] = ShuffleDNS(ConfNet)
 						options[dhcp.OptionRouter] = ShuffleGateway(ConfNet)
 						options[dhcp.OptionDomainName] = []byte(ConfNet.DomainName)
+						if portal.SecureRedirect == "enabled" {
+							if ConfNet.PortalFQDN != "" {
```

Comment:
Could we use the code of detectPortalURL that we move in util/ to build the URL of the portal so that the logic isn't duplicated